### PR TITLE
Log all services to memory

### PR DESCRIPTION
**Pull request by felipecrs**
- All logs can be accessed from `/dev/shm/logs/<service>`, e.g. `/dev/shm/logs/go2rtc/current`.
- NGINX now logs to stdout as well (like in https://github.com/blakeblackshear/frigate/pull/4562)
- Frigate service isn't set in `CMD` anymore as this allows us to log it as well.
    - S6-Overlay does not seem to support customizing logging for CMD
- Refs https://github.com/blakeblackshear/frigate/pull/4562#issuecomment-1332606737
- Is depended on by https://github.com/blakeblackshear/frigate/pull/4562

### CODE REVIEW

**Comment by NickM-27** [comment]
Nice! That should be good

**Comment by felipecrs** [comment]
@NickM-27 the current behavior, which mimics s6-overlay defaults, is to rotate the `current` log every time it reaches 1MB, and keep at most 20 archive files.

As we are logging to memory, we should perhaps consider decreasing the number of archives. Perhaps 1 is enough?

**Comment by NickM-27** [comment]
Yeah I agree that's better archive, might be worth increasing the size a bit to 10MB or so so there's a decent history for users to view.

**Comment by felipecrs** [comment]
Ok, I also agree.

**Comment by felipecrs** [comment]
Done.

**Comment by felipecrs** [comment]
On my tests, this is working. If you could give it a spin as well before merging, it would be nice.

**Comment by NickM-27** [comment]
> On my tests, this is working. If you could give it a spin as well before merging, it would be nice.

Sure, I can do it tomorrow night or Monday morning, I'm out right now

**Comment by felipecrs** [comment]
There's another thing to take into consideration here. Since Frigate isn't run as CMD anymore, s6-overlay now apply some more generic rules to it. Most importantly:

> * `S6_SERVICES_GRACETIME` (default = 3000): How long (in milliseconds) `s6` should wait,
at shutdown time, for services declared in `/etc/services.d` to die before proceeding
with the rest of the shutdown.
> * `S6_KILL_GRACETIME` (default = 3000): How long (in milliseconds) `s6` should wait, at the end of
the shutdown procedure when all the processes have received a TERM signal, for them to die
before sending a `KILL` signal to make *sure* they're dead.

So, do you think 3 seconds is enough for the Frigate process to exit gracefully or should we give it more time?

**Comment by NickM-27** [comment]
I'd say definitely give it more time, don't want it to get killed mid storage maintenance or something.

**Comment by felipecrs** [comment]
Ok, I think there's a bug in s6-overlay which will affect us:

- https://github.com/just-containers/s6-overlay/issues/503

Let's wait for the closure of it before merging this PR.

**Comment by NickM-27** [comment]
> Ok, I think there's a bug in s6-overlay which will affect us:
> 
> * [S6_SERVICES_GRACETIME exits as soon as the first service exits just-containers/s6-overlay#503](https://github.com/just-containers/s6-overlay/issues/503)
> 
> Let's wait for the closure of it before merging this PR.

is it this PR that causes that issue or the update to s6 v3?

**Comment by felipecrs** [comment]
That's a good question. I got to test.

My assumption is that it's this PR, since I'm switching Frigate from CMD to a service.

**Comment by felipecrs** [comment]
I tested here, the same issue happens when using CMD (which means this PR is NOT introducing this issue). Now I'll test now with S6 v2.

**Comment by felipecrs** [comment]
Ok, same issue with s6-overlay v2 which means it was always there. This may actually be the reason for multiple random database corruptions that we found people reporting ([including me](https://github.com/blakeblackshear/frigate/issues/4494)).

Therefore, this PR can be merged as it brings no additional issues.

**Review by NickM-27 — APPROVED** [review]

**Comment by NickM-27** [comment]
how does this work with the dev container? are we still able to start / stop frigate python process as expected and it isn't started automatically?

**Comment by felipecrs** [comment]
@NickM-27 Nothing should change for the devcontainer because of:

https://github.com/blakeblackshear/frigate/pull/4587/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R195-R196

**Comment by NickM-27** [comment]
Thanks @felipecrs 

I'm seeing an error, when I try to view the contents of the log files from the frigate python process it does not have appropriate permissions, the python process will need to be able to read the log file contents.

**Comment by felipecrs** [comment]
Hm... in the devcontainer Python runs as `vscode`. Maybe that's why. But I think it was supposed to have access anyway. I will do some testing tonight, but if you identify some fix feel free to amend this PR.

**Comment by NickM-27** [comment]
> Hm... in the devcontainer Python runs as `vscode`. Maybe that's why. But I think it was supposed to have access anyway. I will do some testing tonight, but if you identify some fix feel free to amend this PR.

that's a fair point, I haven't tested in a full container if python will have access

**Comment by felipecrs** [comment]
Another thing, have in mind that, since the Frigate process is not being supervised in the devcontainer, there won't be any logs for it.

Perhaps we should have a fake Frigate service running in the devcontainer just printing some dummy text, so that the log files gets actually created. What do you think?

**Comment by NickM-27** [comment]
> Another thing, have in mind that, since the Frigate process is not being supervised in the devcontainer, there won't be any logs for it.
> 
> Perhaps we should have a fake Frigate service running in the devcontainer just printing some dummy text, so that the log files gets actually created. What do you think?

Yeah I am aware of that, I'm not sure it's necessary since the logs will be visible in the console (so it isn't needed) and the logs being gone won't cause any real issues. Happy to hear other's opinions though

**Comment by felipecrs** [comment]
@NickM-27 check my last commit. It should fix all issues now.

**Comment by NickM-27** [comment]
Thanks that works great

**Comment by NickM-27** [comment]
@felipecrs I don't think frigate is being started with env variables with this PR

**Comment by NickM-27** [comment]
When I run `printenv` in the container I can see the env variables, but it seems that inside the frigate python process it is unable to see the variables

File: docker/rootfs/etc/services.d/frigate/run [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (felipecrs) [inline docker/rootfs/etc/services.d/frigate/run:1]:
```suggestion
#!/command/with-contenv bash
```

**Comment by felipecrs** [comment]
Thanks. Should work now.

**Pull request merged**